src/tests/system/tests/test_authentication.py

```diff
@@ -0,0 +1,86 @@
+"""
+SSSD Sanity Test Cases
+
+:requirement: offline
+"""
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.topology import KnownTopologyGroup
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.parametrize("method", ["su", "ssh"])
+def test_authentication__login(client: Client, provider: GenericProvider, method: str):
+    """
+    :title: ssh/su login
+    :setup:
+        1. Add user to SSSD
+        2. Set password for user
+        3. Start SSSD
+    :steps:
+        1. Authenticate user with correct password
+        2. Authenticate user with incorrect password
+    :expectedresults:
+        1. User is authenticated
+        2. User is not authenticated
+    :customerscenario: False
+    """
+    provider.user("user1").add(password="Secret123")
+
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password("user1", "Secret123"), "login with correct password failed"
+    assert not client.auth.parametrize(method).password("user1", "NOTSecret123"), "login with wrong password succeeded"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.parametrize("method", ["su", "ssh"])
+def test_authentication__offline_login(client: Client, provider: GenericProvider, method: str):
+    """
+    :title: Offline ssh/su login
+    :setup:
+        1. Add user to SSSD and set its password
+        2. In SSSD domain change "cache_credentials" and "krb5_store_password_if_offline" to "True"
+        3. In SSSD pam change "offline_credentials_expiration" to "0"
+        4. Start SSSD
+    :steps:
+        1. Authenticate user with wrong password
+        2. Authenticate user with correct password
+        3. Make server offline (by removing firewall rules for LDAP, KDC and Global Catalog ports)
+        4. Bring SSSD offline explicitly
+        5. Offline authentication of user with correct password
+        6. Offline authentication of user with wrong password
+    :expectedresults:
+        1. User is not authenticated
+        2. User is authenticated
+        3. Firewall rules dropped
+        4. SSSD is offline
+        5. Offline authentication is successful
+        6. Offline authentication is not successful
+    :customerscenario: False
+    """
+    user = "user1"
```

Comment:
Declaring users/passwords like this this like a waste of time/space.
I would make more sense if the user names were:
a) generated unique for each testcase 
b) declared in a constants file and reused from there.

I would be voting for a) as unique usernames would be easier to find in logs.
@pbrezina WDYT?

Reply:
Generated automatilly or generated hardcoded in the test case? The first one no, as the test would not be reproducible. The second one is in my opinion unneeded hustle, because SSSD logs are cleared before each test so there is no test case overlap in logs (we get logs for each test run).

I personally wouldn't put it in variable though to safe lines.

Reply:
@pbrezina I meant user/group names generated for each test execution. 
Reproducibility if the test in not an issue, we are using fixtures to generate user names at least for ad tests.
Hardcoded names are are problematic when You are debugging a test and You need to kill it midway for some reason. Having different names helps there so even when cleanup failed to happen new iteration of test hass a better chance to run.
I do not remember any code handling /var/log/secure when I was looking at the framework so I still think that having a different username would be beneficial.

Some helper that will generate username starting with a given prefix like this:
```
user = xxx.gen_username("user")
```

Reply:
No, full clean up is the fundamental part of the framework otherwise you start with tainted environment. We won't go this way.

But as discussed, we will clear selected logs and journal so it is fresh as well.